apps/mobile/src/components/ContextMenu/ContextMenu.js

```diff
@@ -0,0 +1,217 @@
+import React, { useMemo } from 'react'
+import { View, Text, TouchableOpacity, ScrollView } from 'react-native'
+import { useTranslation } from 'react-i18next'
+import { useNavigation } from '@react-navigation/native'
+import { PUBLIC_CONTEXT_SLUG, MY_CONTEXT_SLUG, ALL_GROUPS_CONTEXT_SLUG } from '@hylo/shared'
+import { widgetUrl as makeWidgetUrl } from 'util/navigation'
+import FastImage from 'react-native-fast-image'
+import useCurrentUser from '@hylo/hooks/useCurrentUser'
+import useCurrentGroup, { useCurrentGroupSlug } from '@hylo/hooks/useCurrentGroup'
+import useRouteParams from 'hooks/useRouteParams'
+import ContextWidgetPresenter, { orderContextWidgetsForContextMenu } from '@hylo/presenters/ContextWidgetPresenter'
+import useGetContextWidgetChildren from '@hylo/hooks/useGetContextWidgetChildren'
+import useHasResponsibility, { RESP_ADD_MEMBERS, RESP_ADMINISTRATION } from '@hylo/hooks/useHasResponsibility'
+import useLogout from 'hooks/useLogout'
+import WidgetIconResolver from 'components/WidgetIconResolver'
+import GroupMenuHeader from 'components/GroupMenuHeader'
+import { openURL } from 'hooks/useOpenURL'
+import { DEFAULT_APP_HOST } from 'navigation/linking'
+import { URL } from 'react-native-url-polyfill'
+import Loading from 'components/Loading'
+
+export default function ContextMenu () {
+  const { myHome, groupSlug } = useRouteParams()
+  const [{ currentGroup, fetching }] = useCurrentGroup({ setToGroupSlug: groupSlug })
+  const currentGroupSlug = useCurrentGroupSlug()
+  const navigation = useNavigation()
+  const { t } = useTranslation()
+  const contextWidgets = currentGroup?.contextWidgets || []
+  const orderedWidgets = useMemo(() =>
```

Comment:
Let's probably make this a useWidgets hook? The process of trying to do so, or answering to me why not will help us clarify the pattern a lot. I think this pattern is coming from Web, but I am a little confused in the Mobile setup by the use of myHome vs currentGroup. We may not be on Web, but I am fully ready and would prefer to entirely switch to the "Context" naming at this level and do away with the "Group" naming as I think it is obscuring things here. There may be "group" type contexts specific stuff that happens here and certainly below this level, but I think it should be handled semantically as a type of context more clearly. If my mental model is correct, the more we do that here, and I think we can with low effort, the more things will tighten up.

Reply:
myHome is probably redundant now and likely can be stripped out. It existed before the `my` static group was created.

I feel like we've already been on the 'does this need more abstraction' mery-go-round here and changed it into a presenter and then reverted. So, I'd love to punt on this until later, when things are more settled.
The tree-style ordering happens once in mobile, once in web. And the alphabetical ordering happens... once in mobile and once in web.

Reply:
I certainly don't think we're on a merry-go-round, and don't know what you are referring to in terms of something having been abstracted and reverted. In Mobile there has always been a My static group, that is how it was setup originally. I'm not following entirely, but if you're feeling like just pushing through then please make final amendments and I'll give it another careful review and see if it is all coming together. Do you have any changes you will still make?

Reply:
Can you please put a todo comment above this line (or wherever you think it best belongs) to the effect of:

"TODO: myHome is probably redundant now and likely can be stripped out. It existed before the my static group was created."

The rest we'll let ride...